### ISSUE TITLE
Parse error on private class fields

### ISSUE THREAD

**Issue by fxOne** [SonarSource/SonarJS#1117]
**I want to report a bug.**

**SonarJS version:** 5.0<br>
**SonarQube version:** 7.2.1 (build 14109)<br>

**Rule key:** javascript:ParsingError

**Reproducer** 
```
class Counter extends HTMLElement {
  #x = 0; //Parse error
}
```

**Expected behavior**
Private class fields should not throw a parse error. See also Stage 3 Proposal: https://github.com/tc39/proposal-class-fields#private-fields

**Comment by saberduck** [comment]
hello @fxOne, 

thanks for sending us the report. Usually, we don't support stage 3 proposals, we support only released EcmaScript versions. Eventually, we would like to allow to customize our parser (e.g. use babel), so the users can use whatever new language feature they want, but this is not yet available.

**Comment by vilchik-elena** [comment]
@fxOne Currently we are gradually migrating SonarJS from our own parser to esprima-compatible (first step is done in SonarJS 5.0). I see your concern, but this is exactly why we want to use another parser, it's too painful to support all favours and syntaxes of JS.

We could consider patching our parser if the change is small, but in your case we are talking about significant change, many rules might be effected as well as symbol information.

So I apology for the inconvenience you have and hope that SonarJS will be migrated fully to the third-party parser pretty soon.

**Comment by hcomnetworkers** [comment]
Hello, we have the same problem and currently replace the `#` with `_xx` before our sonar-run with a simple bash script. `find src -type f -name "*.js" -print0 | xargs -0 sed -i -e 's/\(\s\|\.\)#\([a-z]\)/\1_xx\2/g'`

The downside is, that all files modified this way lose their SVN information and all tickets are assigned to the default developer.
Is there any way to trick Sonar in picking up the SVN information before the replacement?

Or could you, as a work-around, replace the `#` yourself? This way no actual support for private methods would need to be implemented until you have moved to the other parser.

**Comment by yesvods** [comment]
Are there any tracking in this issue?
